src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -0,0 +1,595 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Xml.XPath;
+
+using Mono.Cecil;
+
+namespace Mono.Linker.Dataflow
+{
+	class XmlFlowAnnotationSource : IFlowAnnotationSource
```

Comment:
Would not be better to do this in a more generic way and introduce a class which convert XML attributes into a new type which implements ICustomAttribute? That way we can use this mechanism for any attribute in the linker if needed not just for `DynamicallyAccessedMembers` ?

Reply:
This is on me - I asked @tlakollo to implement the bare minimum changes to get the `DynamicallyAccessMembers` working for now. After that we can see what would be the best option to make it more generic.